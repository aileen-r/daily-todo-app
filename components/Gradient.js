import React from "react";
import { View, ScrollView, StyleSheet, TextInput, Dimensions } from "react-native";

import ToDoList from "./ToDoList";

//Should be able to import from Layout.js
const width = Dimensions.get('window').width;

export default class Gradient extends React.Component {
  constructor() {
    super();
    this.state = {
      newToDoItem: ''
    };

    newTodoItemController = toDoLabel => {
      this.setState({
        newToDoItem: toDoLabel
      });
    };
  }
  
  render() {
    return (
      <View style={styles.gradient}>
        <TextInput
          style={styles.input}
          placeholder={'Add an item!'}
          value={this.newToDoItem}
          onChangeText={this.newTodoItemController}
          placeholderTextColor={'#999'}
          returnKeyType={'done'}
          autoCorrect={false}
        />
        <ScrollView>
          <ToDoList />
        </ScrollView>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  gradient: {
    width: width - 20,
    padding: 20,
    marginVertical: 30,
    backgroundColor: "#E4E4E4",
    flex: 1
  },
  input: {
    padding: 20,
    borderBottomColor: "#bbb",
    borderBottomWidth: 1,
    fontSize: 24
  }
});
