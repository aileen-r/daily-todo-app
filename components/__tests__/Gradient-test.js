import React from 'react';

import renderer from 'react-test-renderer';
import Gradient from '../Gradient';

test('renders correctly', () => {
  const tree = renderer.create(<Gradient />).toJSON();
  expect(tree).toMatchSnapshot();
});