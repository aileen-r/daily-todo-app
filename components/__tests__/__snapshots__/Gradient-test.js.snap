// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E4E4E4",
      "flex": 1,
      "marginVertical": 30,
      "padding": 20,
      "width": 730,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    placeholder="Add an item!"
    placeholderTextColor="#999"
    returnKeyType="done"
    style={
      Object {
        "borderBottomColor": "#bbb",
        "borderBottomWidth": 1,
        "fontSize": 24,
        "padding": 20,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
